### PR TITLE
Metadata Equality

### PR DESCRIPTION
**Pull request by helenkzhang**
- Implements `IEquatable`
- Override object `Equals` and `GetHashCode`
- Add `PluginIdentity` property in `PluginMetadata`

### CODE REVIEW

File: src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Core/EnumerableExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Core
+{
+    /// <summary>
+    ///     Provides extension methods for <see cref="IEnumerable{T}"/>.
+    /// </summary>
+    internal static class EnumerableExtensions
```
Comment (mslukebo) [inline src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Core/EnumerableExtensions.cs:13]: Wonder if this should go into SDK?

File: src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Core/Packaging/Metadata/ProcessingSourceMetadata.cs [hunk 2]
```diff
@@ -67,5 +69,62 @@ public sealed class ProcessingSourceMetadata
...
+
+            if (this.AvailableTables != null)
+            {
+                foreach (TableMetadata table in this.AvailableTables)
+                {
+                    result = HashCodeUtils.CombineHashCodeValues(result, table?.GetHashCode() ?? 0);
+                }
+            }
+
+            if (this.SupportedDataSources != null)
```
Comment (mslukebo) [inline src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Core/Packaging/Metadata/ProcessingSourceMetadata.cs:119]: Wonder if we should make a CombineHashCodeValues that takes in an `IEnumerable` that does this null check + loop

File: src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Core/EnumerableExtensions.cs (hunk 1 above)
Comment (helenkzhang) [reply src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Core/EnumerableExtensions.cs:13]: It should. But for now we want to put everything in PluginsSystem folder. We can do a refactor once plugins stuff goes official.

Comment (helenkzhang) [reply src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Core/EnumerableExtensions.cs:13]: Same for the other comment :)

**Review by mslukebo — APPROVED** [review]

**Pull request merged**